src/main/java/com/minecolonies/coremod/client/gui/WindowStash.java

```diff
@@ -0,0 +1,126 @@
+package com.minecolonies.coremod.client.gui;
+
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.controls.*;
+import com.ldtteam.blockout.views.ScrollingList;
+import com.ldtteam.structurize.util.LanguageHandler;
+import com.minecolonies.api.colony.IColonyManager;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.Network;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import com.minecolonies.coremod.network.messages.ChangeDeliveryPriorityMessage;
+import com.minecolonies.coremod.network.messages.ChangeDeliveryPriorityStateMessage;
+import com.minecolonies.coremod.network.messages.OpenInventoryMessage;
+import com.minecolonies.coremod.network.messages.PostBoxRequestMessage;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.WindowConstants.*;
+
+/**
+ * Window for the replace block GUI.
+ */
+public class WindowStash extends AbstractWindowSkeleton implements ButtonHandler
```

Comment:
I would probably not have it have a GUI at all. Only the inventory GUI should be enough.